src/app/core/services/auth.service.ts

```diff
@@ -1,67 +1,68 @@
 import { Injectable } from '@angular/core';
 import { BehaviorSubject, ReplaySubject } from 'rxjs';
+
 import { User } from '../entities/user';
+import { LSConstants, EndpointsConstant } from '../constants';
 import { LocalStorageService } from './local-storage-service';
 import { LoaderBlockService } from '../components/loader-block';
+import { RestService } from './rest.service';
 
 @Injectable()
 export class AuthService {
-  private COURSE_USER: string = 'coursesUser';
-  private COURSE_TOKEN: string = 'coursesToken';
   public userInfo: ReplaySubject<User|Object> = new ReplaySubject();
   public authChanged: BehaviorSubject<boolean> = new BehaviorSubject(this.isAuthenticated());
 
   constructor(
       private localStorageService: LocalStorageService,
-      private loaderBlockService: LoaderBlockService
+      private loaderBlockService: LoaderBlockService,
+      private restService: RestService
   ) {
     this.userInfo.next(this.getUserInfo());
   }
 
-  login(user: User, token: string): void {
+  login(login: string, password: string): void {
     this.loaderBlockService.show();
 
-    this.localStorageService.set(this.COURSE_USER, user);
-    this.localStorageService.set(this.COURSE_TOKEN, token);
-
-    setTimeout(
-        () => {
-          this.authChanged.next(true);
-          this.userInfo.next(user);
-
-          this.loaderBlockService.hide();
+    this.restService.post(EndpointsConstant.AUTH.LOGIN, { login, password })
+        .subscribe(
+            ({ token }) => {
+              this.localStorageService.set(LSConstants.COURSES_TOKEN, token);
+              this.authChanged.next(true);
 
-          console.log('User has logged in!');
+              this.restService.get(EndpointsConstant.AUTH.USER_INFO)
+                  .subscribe(
```

Comment:
In my opinion nested subscribes it not the best solution, it's better to avoid them